mongomock/collection.py

```diff
@@ -826,6 +795,77 @@ def _copy_only_fields(self, doc, fields, container):
                 fields[field] = op
             return doc_copy
 
+    def _subdocitem_merge(self, doc_copy, field, sdocitems, subtract=False):
+        sdoc_copy = doc_copy.setdefault(field, [])
+        if len(sdoc_copy) < len(sdocitems):
+            sdoc_copy.extend([{} if not subtract else {'$EMPTY$': True}
+                              for _ in xrange(len(sdocitems) - len(sdoc_copy))])
+        for (sdoc, sdocitem) in zip(sdoc_copy, sdocitems):
+            if not subtract:
+                sdoc.update(sdocitem)
+            else:
+                if sdoc == {'$EMPTY$': True}:
+                    sdoc.clear()
+                    sdoc.update(sdocitem)
+                else:
+                    exc_keys = set(sdoc.keys()).difference(sdocitem.keys())
+                    for key in exc_keys:
+                        del sdoc[key]
+
+    def _filter_subdocuments(self, doc, fields, container):
+        # if 1 was passed in as the field values, include those fields
+        if list(fields.values())[0] == 1:
+            doc_copy = container()
+            for key in fields:
+                key_parts = key.split('.')
+                subdocument = doc
+                subdocument_copy = doc_copy
+                full_key_path_found = True
+                for (i, key_part) in enumerate(key_parts[:-1]):
```

Comment:
I find it very uncommon to put parentheses around the target-list of the for-loop-statement. Even though that it is syntactically and semantically equivalent most people will write
```python
for i, item in enumerate(iterable):
    ...
```